contracts/PricerRule.sol

```diff
@@ -0,0 +1,403 @@
+pragma solidity ^0.5.0;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./PriceOracleInterface.sol";
+import "./EIP20TokenInterface.sol";
+import "./TokenRules.sol";
+import "./SafeMath.sol";
+
+
+contract PricerRule is Organized {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Events */
+
+    event PriceOracleAdded(
+        address _priceOracle
+    );
+
+    event PriceOracleRemoved(
+        address _priceOracle
+    );
+
+    event AcceptanceMarginSet(
+        bytes3 _quoteCurrencyCode,
+        uint256 _acceptanceMargin
+    );
+
+    event AcceptanceMarginRemoved(
+        bytes3 _quoteCurrencyCode
+    );
+
+
+    /* Storage */
+
+    /**
+     * @dev The code of the base currency of the economy.
+     */
+    bytes3 public baseCurrencyCode;
+
+    /**
+     * @dev EIP20 token address of the economy.
+     */
+    EIP20TokenInterface public eip20Token;
+
+    /**
+     * @dev Conversion rate from the base currency to the economy token.
+     */
+    uint256 public conversionRateFromBaseCurrencyToToken;
+
+    /**
+     * @dev Conversion rate decimals from the base currency to the
+     *      economy token.
+     */
+    uint256 public conversionRateDecimalsFromBaseCurrencyToToken;
+
+    /**
+     * @dev Token rules address of the economy.
+     */
+    TokenRules public tokenRules;
+
+    /**
+     * @dev Mapping from pay-currency code to corresponding price oracle.
+     *      The mapped price oracle's base currency code should be the economy
+     *      base currency code (see baseCurrencyCode storage variable)
+     *      and the quote currency code should be the corresponding pay-currency
+     *      code.
+     */
+    mapping(bytes3 => PriceOracleInterface) public baseCurrencyPriceOracles;
+
+    /**
+     * @dev Mapping from pay-currency code to price difference accepatance
+     *      margin. During a pay operation an intended price for
+     *      pay-currency is presented, that is checked against current price of
+     *      pay-currenency wrt stored acceptance margin.
+     */
+    mapping(bytes3 => uint256) public baseCurrencyPriceAcceptanceMargins;
+
+
+    /* Special Functions */
+
+    /**
+     * @notice Constructs a new pricer object.
+     *
+     * @dev Function requires:
```

Comment:
General remark regarding the use of `@dev`: I want to note that the information listed under `@dev` is not relevant to developers, as they can see the code easily, but rather to users of the function. Thus I would suggest putting it to `@notice` instead: https://github.com/ethereum/wiki/wiki/Ethereum-Natural-Specification-Format#tags